rasa/core/training/story_reader/yaml_story_reader.py

```diff
@@ -341,31 +383,57 @@ def _parse_checkpoint(self, step: Dict[Text, Any], is_rule_data: bool) -> None:
 
         self._add_checkpoint(checkpoint_name, slots_dict)
 
-    @staticmethod
-    def is_yaml_story_file(file_path: Text) -> bool:
-        """Check if file contains Core training data or rule data in YAML format.
 
-        Args:
-            file_path: Path of the file to check.
+class StoryParser(YAMLStoryReader):
```

Comment:
@ricwo As we spoke about this refactoring (https://refactoring.guru/replace-conditional-with-polymorphism) a while ago. I had to add more `if`s and decided to give it a shot. Turned out to go relatively smooth.

Reply:
👍